cluster/src/main/java/io/scalecube/cluster/metadata/MetadataCodec.java

```diff
@@ -0,0 +1,10 @@
+package io.scalecube.cluster.metadata;
+
+import java.nio.ByteBuffer;
+
+public interface MetadataCodec<T> {
+
+  T deserialize(ByteBuffer buffer);
+
+  ByteBuffer serialize(T metadata);
+}
```

Comment:
Let's split this into two interfaces. 
`MetadataEncoder.encode() : LOCAL_T -> BB `
and  
`MetadataDecoder.decode() : BB -> REMOTE_T`